### PR TITLE
tests: improve snap-env test

### PR DESCRIPTION
**Pull request by mvo5**
While this was really well covered with unit tests, a proper integration test was missing.

### CODE REVIEW

File: tests/lib/snaps/test-snapd-tools/meta/snap.yaml [hunk 1]
```diff
@@ -15,6 +17,8 @@ apps:
         command: bin/head
     env:
         command: bin/env
+        environment:
+          EXTRA_LOCAL: extra-local
```
Comment (zyga) [inline tests/lib/snaps/test-snapd-tools/meta/snap.yaml]: 💟 

I didn't know this was wired end-to-end :)

File: tests/main/snap-env/task.yaml [hunk 2]
```diff
@@ -10,7 +10,8 @@ debug: |
 execute: |
     echo "Collect SNAP and XDG environment variables"
     test-snapd-tools.env | egrep '^SNAP_' | egrep -v '^SNAP_DID_REEXEC'| sort > snap-vars.txt
-    test-snapd-tools.env | egrep '^XDG_' | sort > xdg-vars.txt 
+    test-snapd-tools.env | egrep '^XDG_' | sort > xdg-vars.txt
+    test-snapd-tools.env | egrep '^EXTRA_' | sort > extra-vars.txt
```
Comment (zyga) [inline tests/main/snap-env/task.yaml]: I have a some extra tests in #2765 but I'll gladly de-conflict whatever lands second.

File: tests/main/snap-env/task.yaml [hunk 3]
```diff
@@ -29,3 +30,8 @@ execute: |
     echo "Enure that XDG environment variables are what we expect"
     egrep -q '^XDG_RUNTIME_DIR=/run/user/0/snap.test-snapd-tools$' xdg-vars.txt
     test $(wc -l < xdg-vars.txt) -ge 1
+
+    echo "Enure that EXTRA environment variables are what we expect"
+    egrep -q '^EXTRA_GLOBAL=extra-global' extra-vars.txt
+    egrep -q '^EXTRA_LOCAL=extra-local' extra-vars.txt
+    test $(wc -l < extra-vars.txt) -ge 2
```
Comment (zyga) [inline tests/main/snap-env/task.yaml]: I'd use `-eq` here but just a nitpick.

**Comment by sergiusens** [comment]
A very common use case is to do
```yaml
environment:
    XDG_CACHE_DIR: $SNAP_USER_DATA/.cache
```

can we make sure that works too?

File: tests/main/snap-env/task.yaml (hunk 2 above)
Comment (mvo5) [inline tests/main/snap-env/task.yaml]: Yeah, if the other lands first, I will de-conflict. Should be pretty trivial.

**Comment by mvo5** [comment]
@sergiusens Your suggestion is implemented in https://github.com/snapcore/snapd/pull/2769/files

**Pull request merged**